compiler/src/dmd/vsoptions.d

```diff
@@ -203,7 +205,11 @@ extern(C++) struct VSOptions
             cmdbuf.writestring(r"\cl.exe");
             return cmdbuf.extractChars();
         }
-        return "cl.exe";
+        //Display an error message if path to cl.exe not found
+        error(Loc.initial, "Error: cl.exe not found. Please ensure that Visual Studio Build Tools are installed and properly configured.");
+        fatal();
```

Comment:
This is inconsistent with `linkerPath()` above. There's no way to 'catch' the error, so no way of using `compilerPath()` to *check* for the existence of `cl.exe`, and falling back to something else if not found.

So IMO the check in `cpreprocess.d` alone is fine, no need to change `vsoptions.d` too.